### PR TITLE
#1538 Add topic page end-to-end tests

### PR DESCRIPTION
**Pull request by wu87988622**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #1538 )
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/cypress/integration/workspacesPage.test.js [hunk 1]
```diff
@@ -20,6 +20,13 @@ import * as utils from '../utils';
 describe('WorkspacesPage', () => {
   before(() => {
     cy.deleteAllWorkers();
+    cy.createWorker();
+  });
+
+  beforeEach(() => {
+    cy.server();
+    cy.route('GET', 'api/workers').as('getWorkers');
+    cy.route('GET', 'api/topics').as('getTopics');
```
Comment (konekoya) [inline ohara-manager/client/cypress/integration/workspacesPage.test.js]: I'm still not sure about this. So we cannot place this in `before` hook?

File: ohara-manager/client/cypress/integration/workspacesPage.test.js [hunk 2]
```diff
@@ -76,4 +83,51 @@ describe('WorkspacesPage', () => {
 
     cy.getByText(clusterName).should('have.length', 1);
   });
+
+  it('add new topic', () => {
```
Comment (konekoya) [inline ohara-manager/client/cypress/integration/workspacesPage.test.js]: _adds__ a new topic

File: ohara-manager/client/cypress/integration/workspacesPage.test.js [hunk 3]
```diff
@@ -76,4 +83,51 @@ describe('WorkspacesPage', () => {
...
+      .getByPlaceholderText('3')
+      .type(1)
+      .getByText('Save')
+      .click()
+      .wait('@getTopics')
+      .getByText(topicName)
+      .should('have.length', 1);
+  });
+
+  it('delete topic', () => {
```
Comment (konekoya) [inline ohara-manager/client/cypress/integration/workspacesPage.test.js]: deletes a topic

File: ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/Overview/OverviewConnectors.js [hunk 4]
```diff
@@ -89,18 +88,8 @@ OverviewConnectors.propTypes = {
...
-          version: PropTypes.shape({
-            key: PropTypes.string.isRequired,
-            defaultValue: PropTypes.oneOfType[PropTypes.string],
-          }).isRequired,
-          author: PropTypes.shape({
-            key: PropTypes.string.isRequired,
-            defaultValue: PropTypes.oneOfType[PropTypes.string],
-          }).isRequired,
+          key: PropTypes.string.isRequired,
+          defaultValue: PropTypes.string,
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesDetailPage/Overview/OverviewConnectors.js]: My bad 😅

File: ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesNewModal/WorkspacesNewModal.js [hunk 5]
```diff
@@ -54,8 +54,7 @@ class WorkerNewModal extends React.Component {
   };
 
   createServices = async values => {
-    const { clientPort, nodeNames } = values;
```
Comment (konekoya) [inline ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesNewModal/WorkspacesNewModal.js]: Isn't the clientPort should come from the users? Am I missing something here?

Comment (wu87988622) [reply ohara-manager/client/src/components/pages/WorkspacesPage/WorkspacesNewModal/WorkspacesNewModal.js]: Now RESTful APIs update(#1498 )
ClientPort cannot be reused.

File: ohara-manager/client/cypress/integration/workspacesPage.test.js (hunk 1 above)
Comment (wu87988622) [reply ohara-manager/client/cypress/integration/workspacesPage.test.js]: Because all it need wait worker or topic, so use beforeEach

Comment (konekoya) [reply ohara-manager/client/cypress/integration/workspacesPage.test.js]: Hmm, does placing these inside the `before` hook work the same as in `beforeEach`?

**Pull request merged**